app_server/public/js/modules/components/datadisplay.js

```diff
@@ -0,0 +1,49 @@
+const m = require('mithril');
+
+const ItemsCount = require('../components/itemscount');
+const Pagination = require('../components/pagination');
+const Datatable = require('../components/datatable');
+
+const Datadisplay = module.exports = {};
+
+Datadisplay.controller = function () {
+  let ctrl = this;
+
+  ctrl.currentPage = m.prop(m.route.param('page') || 1);
+  ctrl.itemsPerPage = m.prop(m.route.param('items') || 10);
+  ctrl.maxPage = m.prop(1);
+
```

Comment:
They're default values.
I don't believe there's any reason to give "1" a name in this case; listings quite naturally start at page 1 (or 0).

Items per page could be extracted, though.
Maybe I should pull `const ITEMS_PER_PAGE = [10, 30, 50];` out from `itemscount.js` and pass it in from here while I'm at it.